libs/features/personalization/personalization.js

```diff
@@ -1058,14 +1075,30 @@ export async function init(enablements = {}) {
       experiments: [],
       geoPrefix: config.locale?.prefix.split('/')[1]?.toLowerCase() || 'en-us',
     };
-    manifests = manifests.concat(await combineMepSources(pzn, promo, mepParam));
-    manifests?.forEach((manifest) => {
-      if (manifest.disabled) return;
-      const normalizedURL = normalizePath(manifest.manifestPath);
-      loadLink(normalizedURL, { as: 'fetch', crossorigin: 'anonymous', rel: 'preload' });
-    });
+
+    manifestPromises = manifestPromises
+      .concat(await combineMepSources(pzn, promo, promoUtilsPromise, mepParam));
   }
 
+  manifests = await Promise.all((await Promise.all(manifestPromises))
```

Comment:
Is the double `Promise.all` there for any specific reason, or is that an omission?

```js
// Untested demo code
manifests = await Promise.all(
  manifestPromises.map(async (promise) => {
    try {
      const resp = await promise;
      if (!resp.ok) {
        if (resp.status === 404) {
          throw new Error('File not found');
        }
        throw new Error(`Invalid response: ${resp.status} ${resp.statusText}`);
      }
      const manifestData = await resp.json();
      return { manifestData, manifestPath: resp.url };
    } catch (e) {
      console.log(`Error loading content: ${resp.url}`, e.message || e);
      return null;
    }
  })
).then(results => results.filter(Boolean));
```